include/net/coap.h

```diff
@@ -360,6 +362,13 @@ u8_t *coap_next_token(void);
 static inline u16_t coap_next_id(void)
 {
 	static u16_t message_id;
+	static int seeded;
+
+	/* According to CoAP specification, message id should be initialized to
+	 * a random number, not to zero. */
+	while (!atomic_cas(&seeded, 0, 1)) {
+		message_id = (u16_t)sys_rand32_get();
```

Comment:
The cast does no harm here so we could leave it, but is not really needed and could be omitted.

Reply:
I tried to be super explicit because I hear MISRA certs are in the pipeline.